modules/core/src/v2/coins/xlm.ts

```diff
@@ -220,13 +223,20 @@ export class Xlm extends BaseCoin {
       throw new Error(`invalid address: ${address}`);
     }
 
+    if (Array.isArray(queryDetails.memoId) && queryDetails.memoId.length > 1) {
+      // valid addresses can only contain one memo id
+      throw new Error(`invalid address '${address}', can only contain one memoId`);
+    }
+
+    const memoId = queryDetails.memoId as string;
+
     try {
-      new BigNumber(queryDetails.memoId);
+      new BigNumber(memoId);
     } catch (e) {
       throw new Error(`invalid address: ${address}`);
     }
 
-    if (!this.isValidMemoId(queryDetails.memoId)) {
+    if (!this.isValidMemoId(memoId)) {
```

Comment:
Another thought would be to move all logic from lines 231 up until here into `isValidMemoId` ... would keep this parent fcn cleaner and consolidate memo validation to a single, testable method.

Just a thought - no need to make sweeping changes though.

Reply:
Yeah, you're right, the try/catch is redundant here. Good catch.